### PR TITLE
Introduce function to claim without passing in signer 

### PR DESCRIPTION
**Pull request by borcherd**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1323,6 +1323,55 @@ async function claimLink({
 	}
 }
 
+/**
+ * Prepares a transaction to claim a link. Broadcasting the transaction is not handled by this function.
+ * @param link: the link to claim
+ * @param recipientAddress: the address to send the link's contents
+ * @param provider: the provider to use for the transaction
+ * @returns
+ */
```
Comment (Hugo0) [inline src/index.ts:1332]: Why is returns empty? Do not have a comment that has  no purpose

File: src/index.ts [hunk 2]
```diff
@@ -1323,6 +1323,55 @@ async function claimLink({
...
+async function prepareClaimTx({
+	link,
+	recipientAddress,
+	provider = undefined,
+}: {
+	recipientAddress: string
+	link: string
+	provider?: ethers.providers.JsonRpcProvider
```
Comment (Hugo0) [inline src/index.ts:1340]: Optional Provider object is how we currently do things, but if an integrator doesn't use ethers, but has private Providers, there should be a way of inputting those (so, an rpc URL string)

Do you know if there's any other place in the code base where we take in an rpc url string? I think there was one place but not sure

File: src/index.ts [hunk 3]
```diff
@@ -1323,6 +1323,55 @@ async function claimLink({
...
+	const contractVersion = params.contractVersion
+	const depositIdx = params.depositIdx
+	const password = params.password
+
+	if (!provider) {
+		provider = await getDefaultProvider(chainId)
+	}
+
+	const keys = generateKeysFromString(password) // deterministically generate keys from password
+	const contract = await getContract(chainId, provider, contractVersion) // weird that signer is passed in here
```
Comment (Hugo0) [inline src/index.ts:1353]: Remove comment

File: src/index.ts [hunk 4]
```diff
@@ -1323,6 +1323,55 @@ async function claimLink({
...
+
+	// cryptography
```
Comment (Hugo0) [inline src/index.ts:1355]: Cryptography hahha

File: src/index.ts [hunk 5]
```diff
@@ -1323,6 +1323,55 @@ async function claimLink({
...
+	const claimParams = await signWithdrawalMessage(
+		contractVersion,
+		chainId,
+		contract.address,
+		depositIdx,
+		recipientAddress,
+		keys.privateKey
+	)
+
+	const tx = await contract.populateTransaction.withdrawDeposit(...claimParams)
```
Comment (Hugo0) [inline src/index.ts:1365]: 👍

File: test/live/claimLink.test.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
+import { claimLink, createLink, getDefaultProvider, prepareClaimTx, setFeeOptions, toggleVerbose } from '../../src'
+import { ethers } from 'ethersv5'
+
+const TEST_WALLET_PRIVATE_KEY = process.env.TEST_WALLET_PRIVATE_KEY
+
+describe('claimLink', () => {
+	toggleVerbose()
+
+	test.skip('claim a link', async () => {
```
Comment (Hugo0) [inline test/live/claimLink.test.ts:9]: Why's test skipped

**Review by Hugo0 — APPROVED** [review]

File: src/index.ts (hunk 2 above)
Comment (borcherd) [reply src/index.ts:1340]: Yes, I agree, but dont think we should change this only here. We are using jsonRpcProvider throughout the entire sdk, so only changing this here is a small change. I think if we want to do this properly, we should do it throughout the entire sdk. I've added in todo comment to remind us in the future. Sound good?

**Pull request merged**